### PR TITLE
Update PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by jlind23**
- Add more checkboxes in the checklist in the pull request template
- Add a definition of done section in the feature request template

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -35,6 +41,9 @@ List here all the items you have verified BEFORE sending this PR. Please DO NOT
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
 - [ ] I have added an integration test or an E2E test
+- [ ] Does Endpoint team need to be informed of this change?
+- [ ] Does Profiling team need to be informed of this change?
+- [ ] Does Cloud defend team need to be informed of this change?
```
Comment (pierrehilbert) [inline .github/PULL_REQUEST_TEMPLATE.md:46]:
```suggestion
- [ ] Does Cloud Defend team need to be informed of this change?
```

**Review by pierrehilbert — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (cmacknz) [reply .github/PULL_REQUEST_TEMPLATE.md:46]: I don't think this needs to be in the PR template. If we made it to a PR without checking first it's too late. We need to think about this before we write the code.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -20,6 +20,12 @@ Explain here the changes you made on the PR. Please explain the WHAT: patterns u
 Explain here the WHY, or the rationale/motivation for the changes.
 -->
 
+## What is the definition of done?
```
Comment (cmacknz) [inline .github/PULL_REQUEST_TEMPLATE.md:23]: Wouldn't this just be copied from the issue description?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (jlind23) [reply .github/PULL_REQUEST_TEMPLATE.md]: I removed these and rather added it to the feature request template only

**Pull request merged**